mlos_bench/mlos_bench/storage/sql/schema.py

```diff
@@ -47,7 +47,7 @@ class DbSchema:
     # pylint: disable=too-many-instance-attributes
 
     # Common string column sizes.
-    _ID_LEN = 512
+    _ID_LEN = 256
```

Comment:
What's this change about?

Reply:
some sql stuff me and Eu Jing both have this, I believe kelly too

Reply:
I don't think we need that. I believe this is from the times @eujing was trying to work around my bug in MLOS when I was saving callables in the DB instead of actual values :)

Reply:
Yes, @DelphianCalamity, @yshady and I have been needing this change whenever we use a fresh database and sqlalchemy is trying to create tables for the first time. @motus I think you did partially fix the original issue we had but, this column size is still too large for MySQL-backends.

Summarizing a thread we have in teams with Kelly, the current offending table is `trial_param`.
The DDL is roughly:

```sql
CREATE TABLE trial_param (
  exp_id VARCHAR(512) NOT NULL,
  trial_id INTEGER NOT NULL,
  param_id VARCHAR(512) NOT NULL,
  param_value VARCHAR(1024),
  PRIMARY KEY (exp_id, trial_id, param_id),
  FOREIGN KEY (exp_id, trial_id) REFERENCES trial (exp_id, trial_id)
)
```

The primary key becomes an issue when using standard encodings like utfmb3 (1-3 bytes per char) or utfmb4 (more bytes per char).
(varchar 512 + integer + varchar 512) in these encodings exceeds 3072 bytes (the limit for MySQL keys)

This results in the MySQL error `Specified key was too long; max key length is 3072 bytes` when executing DDL on a new database.

Reply:
Thanks for explaining @eujing